Framework/PythonInterface/plugins/algorithms/SCDAdjustSampleNorm.py

```diff
@@ -0,0 +1,246 @@
+        self.declareProperty(FloatArrayProperty("BinningDim2", [-8.02, 8.02, 401], FloatArrayLengthValidator(3),
+                                                direction=Direction.Input),
+                             "Binning parameters for the 2nd dimension. Enter it as a"
+                             "comma-separated list of values with the"
+                             "format: 'minimum,maximum,number_of_bins'.")
+
+        self.setPropertySettings("Filename", EnabledWhenProperty('InputWorkspaces', PropertyCriterion.IsDefault))
+        self.setPropertySettings("VanadiumFile", EnabledWhenProperty('VanadiumWorkspace', PropertyCriterion.IsDefault))
+        self.setPropertySettings("InputWorkspaces", EnabledWhenProperty('Filename', PropertyCriterion.IsDefault))
+        self.setPropertySettings("VanadiumWorkspace", EnabledWhenProperty('VanadiumFile', PropertyCriterion.IsDefault))
+
+        event_settings = EnabledWhenProperty('OutputAsMDEventWorkspace', PropertyCriterion.IsDefault)
+        self.setPropertyGroup("MinValues", "MDEvent Settings")
+        self.setPropertyGroup("MaxValues", "MDEvent Settings")
+        self.setPropertyGroup("MergeInputs", "MDEvent Settings")
+        self.setPropertySettings("MinValues", event_settings)
+        self.setPropertySettings("MaxValues", event_settings)
+        self.setPropertySettings("MergeInputs", event_settings)
+
+        histo_settings = EnabledWhenProperty('OutputAsMDEventWorkspace', PropertyCriterion.IsNotDefault)
+        self.setPropertyGroup("BinningDim0", "MDHisto Settings")
+        self.setPropertyGroup("BinningDim1", "MDHisto Settings")
+        self.setPropertyGroup("BinningDim2", "MDHisto Settings")
+        self.setPropertySettings("BinningDim0", histo_settings)
+        self.setPropertySettings("BinningDim1", histo_settings)
+        self.setPropertySettings("BinningDim2", histo_settings)
+
+        self.declareProperty(
+            WorkspaceProperty("OutputWorkspace", "", optional=PropertyMode.Mandatory, direction=Direction.Output),
+            doc="Output MDWorkspace in Q-space, name is prefix if multiple input files were provided.")
+
+    def validateInputs(self):
+        issues = dict()
+
+        filelist = self.getProperty("Filename").value
+        vanfile = self.getProperty("VanadiumFile").value
+        input_ws = self.getProperty("InputWorkspaces")
+        van_ws = self.getProperty("VanadiumWorkspace")
+
+        # Make sure files and workspaces aren't both set
+        if len(filelist) >= 1:
+            if not input_ws.isDefault:
+                issues['InputWorkspaces'] = "Cannot specify both a filename and input workspace"
+        else:
+            if input_ws.isDefault:
+                issues['Filename'] = "Either a file or input workspace must be specified"
+
+        if len(vanfile) <= 0 and van_ws.isDefault:
+            issues['VanadiumFile'] = "Either a vanadium file or vanadium workspace must be specified!"
```

Comment:
I think we should be able to support a workflow without any vanadium correction. I don't think this should be required.